spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskService.java

```diff
@@ -163,6 +164,13 @@ public long executeTask(String taskName, Map<String, String> taskDeploymentPrope
 			List<String> commandLineArgs) {
 		Assert.hasText(taskName, "The provided taskName must not be null or empty.");
 		Assert.notNull(taskDeploymentProperties, "The provided runtimeProperties must not be null.");
+
+		if (maxConcurrentExecutionsReached()) {
+			throw new IllegalStateException(String.format(
+				"The maximum concurrent task executions [%d] is at its limit.",
+				taskConfigurationProperties.getMaximumConcurrentTasks()));
```

Comment:
probably want to use the getMaximumConcurrentTasks() private method, just incase someone in the future adds additional functionality for it.

Reply:
I don't follow

Reply:
Probably not an issue.